frappe/cache_manager.py

```diff
@@ -81,3 +85,6 @@ def clear_single(dt):
 		for name in groups:
 			cache.delete_value(name)
 
+	# Clear all document's cache. To clear documents of a specific DocType document_cache should be restructured
+	clear_document_cache()
```

Comment:
Shouldn't this only clear the specific DocType cache? Why clear everything?

Reply:
Because there is no efficient way to remove all documents of a specific DocType